projects/ardin_mb/state_handler.h

```diff
@@ -118,21 +131,23 @@ class StateHandler : public BoBRobotics::FSM<State>::StateHandler
 
     //! Renderer used for ant world
     BoBRobotics::AntWorld::Renderer m_Renderer;
+    BoBRobotics::AntWorld::RenderTarget m_RenderTargetTopDown;
+    BoBRobotics::AntWorld::RenderTarget m_RenderTargetPanoramic;
 
     //! OpenGL video input used for reading image from framebuffer
-    BoBRobotics::Video::OpenGL m_Input;
+    BoBRobotics::AntWorld::RenderTargetInput m_Input;
 
     //! Route handler - implements the various bits of route-regularizing weirdness from original paper
     BoBRobotics::AntWorld::RouteArdin m_Route;
 
-    //! Snapshot processor - implements the strange resizing algorithm from original paper
-    BoBRobotics::AntWorld::SnapshotProcessorArdin m_SnapshotProcessor;
+    //! OpenCV texture wrapper used to render final snapshot
+    OpenCVTexture m_FinalSnapshotTexture;
 
     //! Class for handling rendering of vector field
     VectorField m_VectorField;
 
     //! Ant's current pose
-    BoBRobotics::Pose2<meter_t, degree_t> m_Pose;
+    BoBRobotics::Pose3<meter_t, degree_t> m_Pose;
```

Comment:
How come the change to 3d, btw?

Reply:
![FlyingAnts4](https://user-images.githubusercontent.com/6793242/83277016-626c8900-a1c9-11ea-8503-397ac8bf521e.jpg)

Reply:
:laughing: